### ISSUE TITLE
Option for configuring colors

### ISSUE THREAD

**Issue by pelowski** [desktop/desktop#6795]
**Red and Green colors are so subtle that colorblind folks like myself can't see the difference**
I wish there was a greater color difference between the red and green for deleted vs new lines.  A person that has Deuteranopia will struggle to see the difference between the shades that have been chosen for new vs. deleted lines.  While I'm not completely colorblind (a condition called monochromacy or achromatopsia, which is extremely rare), lighter shades are really problematic for a person like me to see the difference.

Example:
> The difference in these two colors is only discernible when I really, _really_, **_really_** look closely.  It's so subtle that I can only start to tell they're different when I look for the plus/minus and then work my way backwards.  And unfortunately, the plus/minus at the beginning of the line doesn't really stand out much.
https://www.screencast.com/t/RSXXcyJw

**Perhaps other more prominent ways of distinguishing new/deleted lines?**

1. If the colors were configurable I would make a darker shade of red and a darker shade of green so I could tell the difference.  The level that I usually configure is unfortunately an eyesore for most people, but it's enough that it can stand out and be usable for me.

1. And/or it would be great if the minus/plus sign could be made bold or more distinguishable or if there were horizontal black/dark lines separating the new/deleted so I could more easily differentiate lines in another way.

**Comment by billygriffin** [comment]
Hi @pelowski, thanks so much for the details and sorry you're having a hard time with the red/green differentiation for added/deleted lines. This may be a naive question, but I'm curious if the app is more usable when using the Dark Theme. We did quite a bit of iteration on the contrast for that to make the distinction more visually apparent. You can get there via the menu and then `Preferences` -> `Appearance`

**Comment by pelowski** [comment]
Hey Billy, I should've mentioned that yes, I did try that before I reached out to you although I wasn't sure if that was a secondary purpose for the Dark Theme or not. To answer your question:

> No, I didn't think the Dark Theme was that helpful in this regard.  For my eyes at least the differences in the line colors are marginally more recognizable than they are in the light theme.

Although, now that I look at the Light Theme this afternoon (as opposed to when I posted this late last night) I also see that the Light Theme is _marginally_ more recognizable as well.  I think that one component of this being that [Windows 10 dims the screen at night](https://support.microsoft.com/en-us/help/4027563/windows-10-set-your-display-for-night-time) which I prefer when working late.

So, that is a bit of a mitigating factor here.  (Sorry I didn't realize that last night.)

### In Summary

1. Both the light theme and the dark theme are marginally more noticable now at 2pm EST than they were last night.
1. Neither is that obvious to me even now with my partially-broken cones.
1. To eliminate other potential variables that I might have missed, no, I don't have my screen brightness dialed down a ton but haven't spent any time color-calibrating my two monitors only because with my eyes, it's kind of a crap-shoot anyway.

Hopefully that helps answer your question!

**Comment by billygriffin** [comment]
@pelowski Thanks so much. We've briefly discussed the concept of a high-contrast theme as a way to make the app more accessible for folks with visual impairments, but haven't gone into a ton of detail in terms of what that would look like or level of effort. I really appreciate you bringing this top of mind and we'll continue to evaluate how to improve things. I can't promise anything near-term but it's definitely now on our radar.

cc: @desktop/design

**Comment by TheChrisGlass** [comment]
Just chiming in here to say that in dark mode, it's slightly more readable for me, but a higher contrast mode, or just a means to pick our own colors would be greatly appreciated.

**Comment by tidy-dev** [comment]
Just some more context: The reason this doesn't flag on accessibility reviews is because each line is prefaced with a symbol of `+` or `-` such that color is not the only differentiator for added/deleted lines.

**Comment by Sunhat** [comment]
This would still be a very helpful feature for colorblind folk. I think it's quite a shame this was raised 5 years ago and nothing's been done.

**Comment by tidy-dev** [comment]
> quite a shame this was raised 5 years ago

Definitely heard. There has been over [179 fixes and accessibility improvements](https://github.com/desktop/desktop/pulls?q=is%3Apr+accessibility-audits+is%3Aclosed) over the past year and half. We are small team and as such our fixes have to target accessibility issues that are more likely to block a user with a disabilities ability to utilize software (such as having text below 4.5:1 or 3:1 contract levels or relying entirely on color to communicate a difference/selection). 

In terms of this particular issue, instead of custom solution with custom color palettes that has potential for color selection that unintentionally makes things unusable (and then user having a difficult time reset), the likely next step is to allow system high contrast settings to modify GitHub Desktop's colors schemes and/or detect when system high contrast settings are chosen and then apply special css rules to improve contrast and add increased border visibility on known pain points.

**Comment by s-mcm** [comment]
Is this still being planned? My work college has the same issue with the red/green contrast - he cannot tell the difference whatsoever. 

On the website under Settings -> Appearance there are already colour blind friendly options available. Would it be possible these are added? Thank you.
